components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -0,0 +1,191 @@
+            throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Updating Action for Action Type: " + actionType + " and Action Id: " + actionId);
+        }
+        checkIfActionExists(actionId, tenantDomain);
+        return CACHE_BACKED_DAO.updateAction(getActionTypeFromPath(actionType), actionId, action,
+                IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    @Override
+    public void deleteAction(String actionType, String actionId, String tenantDomain) throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Deleting Action for Action Type: " + actionType + " and Action Id: " + actionId);
+        }
+        checkIfActionExists(actionId, tenantDomain);
+        CACHE_BACKED_DAO.deleteAction(getActionTypeFromPath(actionType), actionId,
+                IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    @Override
+    public Action activateAction(String actionType, String actionId, String tenantDomain) throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Activating Action for Action Type: " + actionType + " and Action Id: " + actionId);
+        }
+        checkIfActionExists(actionId, tenantDomain);
+        return CACHE_BACKED_DAO.activateAction(getActionTypeFromPath(actionType), actionId,
+                IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    @Override
+    public Action deactivateAction(String actionType, String actionId, String tenantDomain) throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Deactivating Action for Action Type: " + actionType + " and Action Id: " + actionId);
+        }
+        checkIfActionExists(actionId, tenantDomain);
+        return CACHE_BACKED_DAO.deactivateAction(getActionTypeFromPath(actionType), actionId,
+                IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    @Override
+    public Map<String, Integer> getActionsCountPerType(String tenantDomain) throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Retrieving Actions count per Type.");
+        }
+        return CACHE_BACKED_DAO.getActionsCountPerType(IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    @Override
+    public Action getActionByActionId(String actionId, String tenantDomain) throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Retrieving Action of Action Id: " + actionId);
+        }
+        return CACHE_BACKED_DAO.getActionByActionId(actionId, IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    /**
+     * Get Action Type from path.
+     *
+     * @param actionType Action Type.
+     * @return Action Type.
+     * @throws ActionMgtClientException If an invalid Action Type is given.
+     */
+    private static String getActionTypeFromPath(String actionType) throws ActionMgtClientException {
```

Comment:
Does this needs to be static ? looks like not

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5771/commits/a96c6e8c2f905af6c7fd842a69a4d972997473a1